pyquil/quil.py

```diff
@@ -242,6 +249,87 @@ def inst(self, *instructions: InstructionDesignator) -> "Program":
 
         return self
 
+    def _add_instruction(self, instruction: quil_rs.Instruction):
```

Comment:
It's not a public function, but it's opaque enough at first glance that it's worth a docstring explaining its intent

Reply:
Good idea, done!